File: php/features.php
<?php
declare(strict_types=1);

// WeltNews SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class WeltNewsFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new WeltNewsBaseFeature();
            case "test":
                return new WeltNewsTestFeature();
            default:
                return new WeltNewsBaseFeature();
        }
    }
}

File: py/features.py
# WeltNews SDK feature factory

from feature.base_feature import WeltNewsBaseFeature
from feature.test_feature import WeltNewsTestFeature


def _make_feature(name):
    features = {
        "base": lambda: WeltNewsBaseFeature(),
        "test": lambda: WeltNewsTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
